File: src/test/java/edu/colorado/plv/fixr/SlicingTestMethods.java
package edu.colorado.plv.fixr;

import java.util.Collection;
import java.util.Collections;

import org.junit.Test;

public class SlicingTestMethods extends TestSlicing {
	@Override
	public String getTestClassName() {return "slice.TestSliceMethods";}
	@Override
	public String getResClassName() {return "slice.TestSliceMethodsRes";}

	@Override
	public Collection<String> getPackages() {
		return Collections.singletonList("java.util.Random");
	}

	@Test
	public void t1(){testSlice("testSliceMethods01");}
	
	@Test
	public void t2(){testSlice("testSliceMethods02");}
	
	@Test
	public void t3(){testSlice("testSliceMethods03");}
	
	@Test
	public void t4(){testSlice("testSliceMethods04");}
	
	@Test
	public void t5(){testSlice("testSliceMethods05");}
	
	@Test
	public void t6(){testSlice("testSliceMethods06");}	
}

File: src/test/scala/edu/colorado/plv/fixr/tests/slicing/SlicingTestMethods.scala
package edu.colorado.plv.fixr.tests.slicing 

class SlicingTestMethods extends TestSlicing("./src/test/resources/jimple",
    "slice.TestSliceMethods", "slice.TestSliceMethodsRes") {

  override def getPackages() : List[String] = {
    return List[String]("java.util.Random")
  }

  test("testSliceMethods01") {testSlice("testSliceMethods01")}
  test("testSliceMethods02") {testSlice("testSliceMethods02")}
  test("testSliceMethods03") {testSlice("testSliceMethods03")}
  test("testSliceMethods04") {testSlice("testSliceMethods04")}
  test("testSliceMethods05") {testSlice("testSliceMethods05")}

  /* still not pass */
  ignore("testSliceMethods06") {testSlice("testSliceMethods06")}
}
